### DISCUSSION TITLE
How to get immediate updates on editing DataGridTextColumn cell?

### DISCUSSION THREAD

**Discussion by NoizeDaemon** [AvaloniaUI/Avalonia#12278] [Q&A] [answered]
When editing a `DataGrid` cell, the changes only get updated when I change row or lose focus on the grid (which makes sense for most use cases). In my case, I want to display a preview of the row content, that should react to changes immediately, not waiting for confirmation.

How could I achieve immediate updates (on every character change) like it would be when binding a `TextBox` text to a `TextBlock` text? 

I'm assuming the displayed text in the `DataGrid` is separate from the actual property on the item and only when the (text) change is confirmed it'll update the property on my object in the bound collection  - while as in the `TextBox`/`TextBlock` example I bind directly to the visible text, explaining the immediate updates.

So is there a way to change the update behaviour of `DataGridTextColumn` or should I build my own column template? Should I try binding to the displayed text of a selected row (which has it's own complications from what I read so far)? Or are there better approaches?

I'm thankful for any advice! :)

**Comment by timunie** [comment, marked as answer]
Use DataGridTemplateColumn instead

**Reply by NoizeDaemon** [reply]
Thanks! That did the trick 😊

I also forgot to implement `INotifyPropertyChanged`/`ReactiveObject` on my model again 🙄 I only saw those changes because I was binding to `SelectedItem` for testing 😅

With the model setup correctly, this works:

```
<DataGridTemplateColumn Header="Zeile1">
        <DataGridTemplateColumn.CellTemplate>
                <DataTemplate DataType="models:PersonImport">
	                <TextBlock Text="{Binding Adresszeile1}" VerticalAlignment="Center" HorizontalAlignment="Center" />
                </DataTemplate>
        </DataGridTemplateColumn.CellTemplate>
        <DataGridTemplateColumn.CellEditingTemplate>
                <DataTemplate DataType="models:PersonImport">
	                <TextBox Text="{Binding Adresszeile1}" HorizontalAlignment="Stretch" />
                </DataTemplate>
        </DataGridTemplateColumn.CellEditingTemplate>
</DataGridTemplateColumn>
```
